integrations/clickup/client.py

```diff
@@ -0,0 +1,230 @@
+import asyncio
+import time
+from typing import Any, AsyncGenerator, Optional, List, Dict
+from httpx import HTTPStatusError, Timeout
+from loguru import logger
+from port_ocean.utils import http_async_client
+from port_ocean.utils.cache import cache_iterator_result
+
+TEAM_OBJ = "__team"
+WEBHOOK_EVENTS = [
+    "taskCreated",
+    "taskUpdated",
+    "taskDeleted",
+    "listCreated",
+    "listUpdated",
+    "listDeleted",
+]
+MINIMUM_LIMIT_REMAINING = 20
+DEFAULT_SLEEP_TIME = 30
+
+# Adjust the concurrency level as needed
+SEMAPHORE = asyncio.BoundedSemaphore(10)
+
+
+class ClickupClient:
+    """Clickup client to interact with Clickup API."""
+
+    def __init__(self, clickup_url: str, clickup_token: str, archived: bool):
+        self.clickup_url = clickup_url
+        self.clickup_token = clickup_token
+        self.api_url = f"{self.clickup_url}/api/v2"
+        self.client = http_async_client
+        self.client.timeout = Timeout(60)
+        self.archived = archived
```

Comment:
since this is a bool variable, I would rename it to `is_archived`

Reply:
Done